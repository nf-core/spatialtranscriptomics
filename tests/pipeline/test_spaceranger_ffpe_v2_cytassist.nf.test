nextflow_pipeline {
    name "Test full workflow (incl. Space Ranger v2)"
    script "main.nf"
    tag "pipeline"

    test("Space Ranger FFPE v2 CytAssist") {
        when {
            params {
                // This is the default `test` profile; params are not necessary
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(

                // Workflow
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("nf_core_pipeline_software_mqc_versions.yml") },

                // Data
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/data/adata_processed.h5ad").exists() },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/data/sdata_processed.zarr").exists() },

                // Reports
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/quality_controls.html").text.contains("final results of all the filtering") },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/clustering.html").text.contains("spatial distribution of clusters") },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/spatially_variable_genes.html").text.contains("Spatial transcriptomics data can give insight") },

                // DEGs
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/degs/spatially_variable_genes.csv").exists() },

                // Space Ranger
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/spaceranger/outs/web_summary.html").exists() },
                { assert snapshot(
                    path("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/spaceranger/outs/raw_feature_bc_matrix.h5"),
                    path("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/spaceranger/outs/filtered_feature_bc_matrix.h5"),
                )},
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/spaceranger/outs/raw_feature_bc_matrix.h5").exists() },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/spaceranger/outs/spatial/tissue_positions.csv").exists() },

                // MultiQC
                { assert file("$outputDir/multiqc/multiqc_report.html").text.contains("CytAssist_11mm_FFPE_Human_Glioblastoma_2_4")}
            )
        }
    }
}
