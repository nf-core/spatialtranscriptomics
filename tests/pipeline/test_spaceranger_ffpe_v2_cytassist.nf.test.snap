{
    "nf_core_pipeline_software_mqc_versions.yml": {
        "content": [
            "{CLUSTERING={quarto=1.4.549, papermill=null}, FASTQC={fastqc=0.12.1}, QUALITY_CONTROLS={quarto=1.4.549, papermill=null}, READ_DATA={scanpy=1.9.8}, SPACERANGER_COUNT={spaceranger=2.1.0}, SPACERANGER_UNTAR_REFERENCE={untar=1.3}, SPATIALLY_VARIABLE_GENES={quarto=1.4.549, papermill=null}, UNTAR_SPACERANGER_INPUT={untar=1.3}, Workflow={nf-core/spatialtranscriptomics=v1.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T18:45:01.126974"
    }
}